core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceBillingAccountMonitor.scala

```diff
@@ -53,31 +56,97 @@ class WorkspaceBillingAccountMonitor(dataSource: SlickDataSource, gcsDAO: Google
-      })
-    } yield dbResult
+    } yield ()
+
+    updateGoogleResult.recoverWith {
+      case e: RawlsExceptionWithErrorReport if e.errorReport.statusCode == Option(StatusCodes.Forbidden) && newBillingAccount.isDefined =>
+        val message = s"Rawls does not have permission to set the Billing Account on ${googleProjectId} to ${newBillingAccount.get}"
+        dataSource.inTransaction({ dataAccess =>
+          dataAccess.rawlsBillingProjectQuery.updateBillingAccountValidity(newBillingAccount.get, isInvalid = true) >>
+          dataAccess.workspaceQuery.updateWorkspaceBillingAccountErrorMessages(googleProjectId, s"${message} ${e.getMessage}")
+        }).flatMap { _ =>
+          logger.warn(message, e)
+          Future.failed(e)
+        }
+      case e: Throwable =>
+        dataSource.inTransaction { dataAccess =>
+          dataAccess.workspaceQuery.updateWorkspaceBillingAccountErrorMessages(googleProjectId, e.getMessage)
+        }.flatMap { _ =>
+          Future.failed(e)
+        }
+    }
+  }
+
+  /**
+    * Explicitly sets the Billing Account value on the given Google Project.  Any logic or conditionals controlling
+    * whether this update gets called should be written in the calling method(s).
+ *
+    * @param googleProjectId
+    * @param newBillingAccount
+    */
+  private def setBillingAccountOnGoogleProject(googleProjectId: GoogleProjectId,
+                                               newBillingAccount: Option[RawlsBillingAccountName]): Future[ProjectBillingInfo] =
+    newBillingAccount match {
+      case Some(billingAccount) => gcsDAO.setBillingAccountName(googleProjectId, billingAccount)
+      case None => gcsDAO.disableBillingOnGoogleProject(googleProjectId)
+    }
+
+  /**
+    * Explicitly sets the Billing Account value for all Workspaces that are in the given Project.  Any logic or
+    * conditionals controlling whether this update gets called should be written in the calling method(s).
+    * @param googleProjectId
+    * @param newBillingAccount
+    * @return
+    */
+  private def setBillingAccountOnWorkspacesInProject(googleProjectId: GoogleProjectId,
+                                                     newBillingAccount: Option[RawlsBillingAccountName]): Future[Int] =
+    dataSource.inTransaction({ dataAccess =>
+      dataAccess.workspaceQuery.updateWorkspaceBillingAccount(googleProjectId, newBillingAccount)
+    })
+
+  /**
+    * Gets the Billing Account name out of a ProjectBillingInfo object wrapped in an Option[RawlsBillingAccountName] and
+    * appropriately converts `null` or `empty` String into a None.
+    * @param projectBillingInfo
+    * @return
+    */
+  private def getBillingAccountOption(projectBillingInfo: ProjectBillingInfo): Option[RawlsBillingAccountName] = {
+    if (projectBillingInfo.getBillingAccountName == null || projectBillingInfo.getBillingAccountName.isBlank)
+      None
+    else
+      Option(RawlsBillingAccountName(projectBillingInfo.getBillingAccountName))
   }
```

Comment:
`Option` can encode this:

```suggestion
  private def getBillingAccountOption(projectBillingInfo: ProjectBillingInfo): Option[RawlsBillingAccountName] = 
    Option(projectBillingInfo.getBillingAccountName).filter(_.isBlank).map(RawlsBillingAccountName)
```